projects/plugins/jetpack/sal/class.json-api-site-base.php

```diff
@@ -1,143 +1,391 @@
 		return $this->blog_id;
 	}
 
+	/**
+	 * Returns the site name.
+	 *
+	 * @return string
+	 */
 	public function get_name() {
 		return (string) htmlspecialchars_decode( get_bloginfo( 'name' ), ENT_QUOTES );
 	}
 
+	/**
+	 * Returns the site description.
+	 *
+	 * @return string
+	 */
 	public function get_description() {
 		return (string) htmlspecialchars_decode( get_bloginfo( 'description' ), ENT_QUOTES );
 	}
 
+	/**
+	 * Returns the URL for the current site.
+	 *
+	 * @return string
+	 */
 	public function get_url() {
 		return (string) home_url();
 	}
 
+	/**
+	 * Returns the number of published posts with the 'post' post-type.
+	 *
+	 * @return int
+	 */
 	public function get_post_count() {
 		return (int) wp_count_posts( 'post' )->publish;
 	}
 
+	/**
+	 * A prototype function for get_quota - currently returns null.
+	 *
+	 * @return null
+	 */
 	public function get_quota() {
 		return null;
 	}
 
+	/**
+	 * Returns true if a site has the 'videopress' option enabled, false otherwise.
+	 *
+	 * @see class.json-api-site-jetpack.php for implementation.
+	 */
 	abstract public function has_videopress();
 
+	/**
+	 * Sets the upgraded_filetypes_enabled Jetpack option to true as a default.
+	 *
+	 * @see class.json-api-site-jetpack.php for implementation.
+	 */
 	abstract public function upgraded_filetypes_enabled();
 
+	/**
+	 * Sets the is_mapped_domain Jetpack option to true as a default.
+	 *
+	 * Primarily used in WordPress.com to confirm the current blog's domain does or doesn't match the primary redirect.
+	 *
+	 * @see class.json-api-site-jetpack.php for implementation.
+	 */
 	abstract public function is_mapped_domain();
 
+	/**
+	 * Fallback to the home URL since all Jetpack sites don't have an unmapped *.wordpress.com domain.
+	 *
+	 * @see class.json-api-site-jetpack.php for implementation.
+	 */
 	abstract public function get_unmapped_url();
 
+	/**
+	 * Whether the domain is a site redirect or not. Defaults to false on a Jetpack site.
+	 *
+	 * Primarily used in WordPress.com where it is determined if a HTTP status check is a redirect or not and whether an exception should be thrown.
+	 *
+	 * @see class.json-api-site-jetpack.php for implementation.
+	 */
 	abstract public function is_redirect();
 
+	/**
+	 * Defaults to false on Jetpack sites, however is used on WordPress.com sites.
+	 *
+	 * @see class.json-api-site-jetpack.php for implementation.
+	 */
 	abstract public function is_headstart_fresh();
 
+	/**
+	 * If the site's current theme supports post thumbnails, return true (otherwise return false).
+	 *
+	 * @see class.json-api-site-jetpack-base.php for implementation.
+	 */
 	abstract public function featured_images_enabled();
 
+	/**
+	 * Whether or not the Jetpack 'wordads' module is active on the site.
+	 *
+	 * @see class.json-api-site-jetpack.php for implementation.
+	 */
 	abstract public function has_wordads();
 
+	/**
+	 * Defaults to false on Jetpack sites, however is used on WordPress.com sites.
```

Comment:
I have a few with comments like this. The WordPress.com files don't have comments where these functions have been built out either. It would seem that that is the best place for more descriptive comments initially, but I could fill out the details here instead (and class.json-api-site-jetpack.php where some functions return a default value).

Reply:
Actually I've expanded on these now so they should be better.